cypress-io/cypress
# feat: update to Vite 3

- Closes https://github.com/cypress-io/cypress/issues/22784

### User facing changelog


- Update Cypress monorepo to use Vite 3
- As part of the Vite 3 update, I fixed a bug in our `vite-dev-server` integration that was introduced by a breaking change in Vite 3. The bug was Vite plugins that modify `index.html` were not correctly inserted into the `index.html` (eg `@vitejs/plugin-react`, [example issue](https://github.com/vitejs/vite/issues/9362).

### Additional details


Vite 3 came out, it had some breaking changes. [Breaking changes](https://vitejs.dev/guide/migration.html#advanced). [[#6901] fix: sequential injection of tags in transformIndexHtml](https://github.com/vitejs/vite/pull/6901) broke us, specifically.

It's fixed in this PR, [see here](https://github.com/cypress-io/cypress/pull/22915/files#diff-5b3ea0fa70e7e1e8a968123451e3b68ad11cde664c56ef07e4ccdd02769887e5R56-R94). 

Note: Percy has 51 changes - I actually **fixed** a bug, Percy was capturing the UI in the incorrect state. See [this comment](https://github.com/cypress-io/cypress/pull/22915#discussion_r929427553) on the problem. I will hit approve before we merge, once everyone is happy the PR.

### Steps to test


CI should be passing, and since we use Vite internally a lot, if it's passing, everything should be :100:. 

I'd also recommend pulling down the repo, and doing a `yarn dev` in something like `packages/app` with component testing, and making sure it all works as expected.

If you like, you could try replicate this user issue and make it it works: https://github.com/vitejs/vite/issues/9362 (we have tests for this specific case - using React and the Vite plugin - but good to be certain).


### How has the user experience changed?


Can use Vite 3 with Cypress Component Testing.

### PR Tasks
